### PR TITLE
 Designs page

### PR DESCRIPTION
**Pull request by LostInProgres**
Hello!!! It is I, who still doesn't really know how this page works!!!

pretty much just two pages in the user sidebar- one showing all characters a user is accredited wise in terms of design, and one that shows all characters a user is accredited to in terms of masterlist art.

I don't- know what else to put here. Please have mercy on me. o/

If you saw my previous attempt, no you didn't, please spare me.

### CODE REVIEW

File: app/Models/User/User.php [hunk 1]
```diff
@@ -5,6 +5,7 @@
 use App\Models\Character\Character;
 use App\Models\Character\CharacterBookmark;
 use App\Models\Character\CharacterImageCreator;
+use App\Models\Character\CharacterImage;
```
Comment (itinerare) [inline app/Models/User/User.php:8]: This isn't necessary but should get stripped out by pint whenever it... actually acquiesces to run next...

File: resources/views/user/character_arts.blade.php [hunk 2]
```diff
@@ -0,0 +1,33 @@
+@extends('user.layout')
+
+@section('profile-title')
+    {{ $user->name }}'s Favorites
```
Comment (itinerare) [inline resources/views/user/character_arts.blade.php:4]: Page title needs to be changed appropriately!

File: resources/views/user/character_designs.blade.php [hunk 3]
```diff
@@ -0,0 +1,33 @@
+@extends('user.layout')
+
+@section('profile-title')
+    {{ $user->name }}'s Favorites
```
Comment (itinerare) [inline resources/views/user/character_designs.blade.php:4]: Page title again

File: app/Http/Controllers/Users/UserController.php [hunk 4]
```diff
@@ -317,6 +318,56 @@ public function getUserGallery(Request $request, $name) {
...
+            });
+        });
+
+        if (!Auth::check() || !(Auth::check() && Auth::user()->hasPower('manage_characters'))) {
+            $characters->visible();
+        }
+
+        return view('user.character_arts', [
+            'user'        => $this->user,
+            'designs'      => $characters->get(),
```
Comment (itinerare) [inline app/Http/Controllers/Users/UserController.php:342]: This should probably just be `'characters'` for ease.

File: app/Http/Controllers/Users/UserController.php [hunk 5]
```diff
@@ -317,6 +318,56 @@ public function getUserGallery(Request $request, $name) {
...
+            });
+        });
+
+        if (!Auth::check() || !(Auth::check() && Auth::user()->hasPower('manage_characters'))) {
+            $characters->visible();
+        }
+
+        return view('user.character_designs', [
+            'user'        => $this->user,
+            'designs'      => $characters->get(),
```
Comment (itinerare) [inline app/Http/Controllers/Users/UserController.php:367]: Likewise here.

File: resources/views/user/character_arts.blade.php (hunk 2 above)
Comment (AW0005) [inline resources/views/user/character_arts.blade.php:4]: Pretty sure this should say something else

File: resources/views/user/character_arts.blade.php [hunk 6]
```diff
@@ -0,0 +1,33 @@
...
+@endsection
+
+@section('profile-content')
+{!! breadcrumbs(['Users' => 'users', $user->name => $user->url, 'Character Designs' => $user->url . '/designs']) !!}
```
Comment (itinerare) [inline resources/views/user/character_arts.blade.php:8]: Breadcrumbs need to be updated for art page.

File: resources/views/user/character_arts.blade.php [hunk 7]
```diff
@@ -0,0 +1,33 @@
...
+
+    <h1>
+        Designs
+    </h1>
+
+    <div class="row">
+        @foreach ($designs as $character)
```
Comment (itinerare) [inline resources/views/user/character_arts.blade.php:15]: `$designs` > `$characters` if following the above comment.

File: resources/views/user/character_designs.blade.php [hunk 8]
```diff
@@ -0,0 +1,33 @@
...
+
+@section('profile-content')
+{!! breadcrumbs(['Users' => 'users', $user->name => $user->url, 'Character Designs' => $user->url . '/designs']) !!}
+
+    <h1>
+        Designs
+    </h1>
+
+    <div class="row">
+        @foreach ($designs as $character)
```
Comment (itinerare) [inline resources/views/user/character_designs.blade.php:15]: `$designs` > `$characters` if following the above comment.

**Review by AW0005 — COMMENTED** [review]
Is there anything majorly different between the arts and designs blades? 🤔 Wondering if you could just have one blade file you use for both.

File: resources/views/user/character_arts.blade.php [hunk 9]
```diff
@@ -0,0 +1,33 @@
...
+
+@section('profile-title')
+    {{ $user->name }}'s Favorites
+@endsection
+
+@section('profile-content')
+{!! breadcrumbs(['Users' => 'users', $user->name => $user->url, 'Character Designs' => $user->url . '/designs']) !!}
+
+    <h1>
+        Designs
```
Comment (itinerare) [inline resources/views/user/character_arts.blade.php:11]: Needs to be changed for art page.

**Review by itinerare — CHANGES_REQUESTED** [review]
Actually pretty well in order already! Just some minor tweaks that need making.

**Comment by itinerare** [comment]
> Is there anything majorly different between the arts and designs blades? 🤔 Wondering if you could just have one blade file you use for both.

Yeah this is probably the actual move, though minorly more complicated.

**Review by itinerare — APPROVED** [review]

**Review by AW0005 — COMMENTED** [review]
The `resources/views/user/character_arts.blade.php` should probably be deleted since it's not being used now 🤔

**Comment by itinerare** [comment]
~I have the power~

**Review by AW0005 — APPROVED** [review]

**Pull request merged**